sdk/identity/azure-identity/tests/managed-identity-live/appservice.md

```diff
@@ -0,0 +1,190 @@
+```sh
+az keyvault set-policy -n $KEY_VAULT_NAME \
+    --object-id $(az identity show -g $RESOURCE_GROUP -n $MANAGED_IDENTITY_NAME --query principalId -o tsv) \
+    --secret-permissions set delete
+```
+
+## Web app: user-assigned identity
+```sh
+az webapp create -n $WEB_APP_USER_ASSIGNED -g $RESOURCE_GROUP --plan $APP_SERVICE_PLAN --runtime "python|3.6"
+```
+
+Set app configuration:
+```sh
+az webapp config appsettings set -g $RESOURCE_GROUP -n $WEB_APP_USER_ASSIGNED \
+  --settings AZURE_IDENTITY_TEST_VAULT_URL=$(az keyvault show -g $RESOURCE_GROUP -n $KEY_VAULT_NAME --query properties.vaultUri -o tsv) \
+   AZURE_IDENTITY_TEST_MANAGED_IDENTITY_CLIENT_ID=$(az identity show -g $RESOURCE_GROUP -n $MANAGED_IDENTITY_NAME -o tsv --query clientId)
+```
+
+At the time of writing, attaching user-assigned identities is impossible through the Azure CLI
+([Azure/azure-cli#9887](https://github.com/Azure/azure-cli/issues/9887)).
+Use the Azure Portal to attached the managed identity created above to the Web App (see
+[App Service documentation](https://docs.microsoft.com/en-us/azure/app-service/overview-managed-identity?tabs=dotnet#adding-a-user-assigned-identity)).
+
+# Build the Docker image
+The test are deployed as a container. The following command lines assume this working directory:
+> `azure-sdk-for-python/sdk/identity/azure-identity/tests`
+
+### Authenticate to ACR
+```sh
+az acr login -n $ACR_NAME
+```
+
+### Set a variable for the image name
+```sh
+export IMAGE_NAME=$(az acr show -n $ACR_NAME --query loginServer -o tsv)/webapp-managed-id-test \
+    PYTHON_VERSION=2.7
+```
+
+### Build the image
+```sh
+docker build --no-cache --build-arg PYTHON_TAG=$PYTHON_VERSION -t $IMAGE_NAME:$PYTHON_VERSION ./managed-identity-live
+```
+
+### Push it to the registry
+```sh
+docker push $IMAGE_NAME:$PYTHON_VERSION
+```
+
+Then set `PYTHON_VERSION` to the latest 3.x and run the above `docker build`
+and `docker push` commands again. (It's safe--and faster--to omit
+`--no-cache` from `docker build` the second time.)
+
+# Deploy test code
+Configure the Web Apps to use the image. For example, for the app using system-assigned identity:
+```sh
+az webapp config container set -g $RESOURCE_GROUP -n $WEB_APP_SYSTEM_ASSIGNED \
+  -i $IMAGE_NAME:$PYTHON_VERSION \
+  -r $(az acr show -n $ACR_NAME --query loginServer -o tsv) \
+  -p $(az acr credential show -n $ACR_NAME --query "passwords[0].value" -o tsv) \
+  -u $(az acr credential show -n $ACR_NAME --query username -o tsv)
+```
+Do this again for the app using a user-assigned identity (replace `WEB_APP_SYSTEM_ASSIGNED` with `WEB_APP_USER_ASSIGNED`).
+
+# Run tests
+For each Web App, get the tests' invocation URLs, and browse to each. For example, for the app using system-assigned identity:
+```sh
+curl https://$WEB_APP_SYSTEM_ASSIGNED.azurewebsites.net
+```
+Do this again for the app using a user-assigned identity (replace `WEB_APP_SYSTEM_ASSIGNED` with `WEB_APP_USER_ASSIGNED`).
```

Comment:
We need to test system- and user-assigned identity on 2.7 and latest 3.x. So, four runs required.

Reply:
That was a comment, not a suggested addition 😄. What I meant is, reading this it wasn't clear to me that the tester must run the tests four times--2.7 and 3.x once each, on both web apps.

Reply:
I know and adding this can clarify that, not true? 😄

Reply:
I imagine it would be confusing to others coming to the document without the right context, and by itself it leaves some important details implicit. Testing a second Python version means running another container, which means updating the webapp's configuration, right?